core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1525,10 +1539,25 @@ private[spark] class DAGScheduler(
             s" ${task.stageAttemptId} and there is a more recent attempt for that stage " +
             s"(attempt ${failedStage.latestInfo.attemptNumber}) running")
         } else {
+          if (!event.reason.asInstanceOf[FetchFailed].countTowardsStageFailures) {
```

Comment:
Can we have a comment describing what the intended behaviour is? Back to the questions around retrying I'm not sure this is the right thing to do, but maybe I just don't understand yet.

Reply:
I have tried to explain this here
#27636 (comment)

Reply:
countTowardsStageFailures is introduced as the part of this PR - So this is for the decommission nodes only. Below is the code where we are setting the value countTowardsStageFailures=false 

```
// Do account fetch failure exception raised by decommissioned
          // node against stage failure. Here the logic is to specify,
          // if the task failed due to fetchFailed of decommission nodes than
          // don't count towards the stageFailure. countTowardsStageFailures
          // variable of TaskEndReason, that can be used in DAG scheduler to account
          // fetch failure while checking the stage abort
          decommissionTracker match {
            case Some(decommissionTracker) =>
              if (decommissionTracker.isNodeDecommissioned(fetchFailed.bmAddress.host)) {
                logInfo(s"Do not count fetch failure from decommissioned" +
                  s" node ${fetchFailed.bmAddress.host}")
                fetchFailed.countTowardsStageFailures = false
              }
            case _ =>
            // No action
          }
```

Reply:
oh sorry, I misread it as countTowardsTaskFailures which is pre-existing.  Right now the logic for this is very specific to decommision, so think it should be renamed to indicate so.  If there are other cases this variable could be useful - then the question is can you differenitate those from the decomission case so you can still do this? If not just rename it to be specific to decomission

Reply:
updated the name of the variable here